projects/base_project.py

```diff
@@ -34,11 +42,43 @@ def class_mapping(self, results1, results2):
     def create_proj_save_dir(self, dir_name):
         """
         See ibase_project.py
-
-        Returns:
-            None
         """
         _cur_dir = pdirname(pabspath(__file__))
         self.proj_dir = pjoin(_cur_dir, f'../data/{dir_name}')
         self.proj_dir = pabspath(self.proj_dir)  # normalise the link
         print(f'1. Created/Found project folder under {self.proj_dir} path')
+
+    def __connect_models__(self):
+        """
+        Initializes the YOLO models and connects them to the appropriate device (CPU or GPU).
+
+        Returns:
+            tuple: A tuple containing two YOLO models.
+
+        Raises:
+            ModuleNotFoundError: If the models cannot be loaded.
+        """
+        _cur_dir = os.getcwd()
+        # initialise the yolo model, additionally use the device parameter to specify the device to run the model on.
+        self.device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        _cur_dir = pdirname(pabspath(__file__))
+        model_dir = pjoin(_cur_dir, f'../models')
+        model_dir = pabspath(model_dir)  # normalise the link
+
+        if not self._var.MODEL_NAME:
+            raise ModuleNotFoundError('Model not found!')
+
+        model = YOLO(pjoin(model_dir, self._var.MODEL_NAME)).to(self.device)
```

Comment:
Models should go into project class, not in base project.

Reply:
Refactored